File: Payhub - SDK/php_SDK/Samples/RecurringBillUpdateSample.php
<?php
/* The current url, oauth_token, orgId and Terminal Id provided in this example, are only for testing purposes
*  For development purposes you need to contact the Payhub Integration Support team. They will provide you with  *   all you need.
*  Thanks.
*/
$path_to_IncludeClases="../com/payhub/ws/extra/includeClasses.php";
include_once $path_to_IncludeClases;

//Defining the Web Service URL
//$WsURL="https://sandbox-api.payhub.com/api/v2/";
//$oauth_token = "2a5d6a73-d294-4fba-bfba-957a4948d4a3";
$WsURL="http://localhost:8251/payhubws/api/v2/";
$oauth_token = "43d50993-cc59-4f01-8e93-a53e0e2e59c0";

//Defining data for the SALE transaction
// Merchant data (obtained from the payHub Virtual Terminal (3rd party integration)
$merchant = new Merchant();
$merchant->setOrganizationId(10002);
$merchant->setTerminalId(2);

$bill= new Bill();
$bill->setBaseAmount(1.00);
//Credit card data
$card_data = new CardData();
$card_data->setCardNumber("4012881888818888");
$card_data->setCardExpiryDate("202012");
$card_data->setBillingZip("60527");
// Customer data
$customer = new Customer();
$customer->setFirstName("Joe");
$customer->setLastName("Tester");
$customer->setPhoneNumber("844-217-1631");
$customer->setPhoneType("M");

$montly_s = new MontlySchedule();
$montly_s->monthly_type="E";
$montly_s->monthly_each_days=array(1,15);

$start=new DateTime("2016-4-29");
$start=$start->format('Y-m-d');
$type="O";
$endDate=new DateTime("2016-8-29");
$endDate=$endDate->format('Y-m-d');
$scheduleSandE=new ScheduleSartAndEnd();
$scheduleSandE->start_date=$start;
$scheduleSandE->end_date_type=$type;
$scheduleSandE->end_date=$endDate;

$schedule =new Schedule("M");
$schedule->monthly_schedule=$montly_s;
$schedule->schedule_start_and_end=$scheduleSandE;
$schedule->bill_generation_interval=1;

$recurringBill=new RecurringBill($merchant,$customer,$bill,$card_data,$schedule);
$transaction = new TransactionManager($merchant,$WsURL,$oauth_token);
$response = $transaction->doRecurringBill($recurringBill);

if($response->getErrors()==null){
    $transactionId = $response->getLastRecurringBillResponse()->getRecurringBillId();
    $bill->setBaseAmount(2.00);
    $recurringBill2 = new RecurringBill($bill);
    $canUpdate=$transaction->updateRecurringBill($transactionId,$recurringBill2);
    if(!is_array($canUpdate)){
        $statusUpdated = $transaction->getRecurringBillInformation($transactionId);
        var_dump($statusUpdated);
        if($statusUpdated->getErrors()==null){
            var_dump($statusUpdated->getBillInformation());
        }
    }else{
        var_dump($canUpdate);
    }
}else{
    var_dump($response->getErrors());
}

File: Payhub - SDK/ruby_SDK/Samples/recurring_bill_update_sample.rb
require 'PayHubSDK/com/payhub/ws/extra/include_classes'

wsURL="https://staging-api.payhub.com/api/v2/"
oauth_token = "107d74ab-4a18-4713-88ff-69bd05710086"

merchant = Merchant.new
merchant.organization_id=10127
merchant.terminal_id=215


# bill data
bill= Bill.new
bill.base_amount=1.00

#Credit card data
card_data = CardData.new
card_data.card_number="4055011111111111"
card_data.card_expiry_date="202012"
card_data.billing_zip="99997-0003"
# Customer data
customer = Customer.new
customer.first_name="First"
customer.last_name="Contact"
customer.phone_number="844-217-1631"
customer.phone_type="W"

montly_s=MonthlySchedule.new("E")
start=Date.new(2016,8,29)
montly_s.monthly_each_days=Array.new
montly_s.monthly_each_days.push(15)
type="O"
endDate=Date.new(2017,8,29)
scheduleSandE=ScheduleStartAndEnd.new(start,type,endDate)
schedule = Schedule.new("M")
schedule.bill_generation_interval=1
schedule.monthly_schedule=montly_s
schedule.schedule_start_and_end=scheduleSandE



transaction = TransactionManager.new(wsURL,oauth_token,merchant)
recurringBill = RecurringBill.new(merchant,customer,bill,card_data,schedule)
response = transaction.doRecurringBill(recurringBill)
if response.errors==nil
  id=response.lastRecurringBillResponse.recurringBillId
  bill.base_amount=2.00
  recForUpdate=RecurringBill.new()
  recForUpdate.bill=bill
  canUpdate = transaction.updateRecurringBill(id,recForUpdate)
  #if can update is an array, then there is an error, if not, the update was successful
  if canUpdate.is_a?(Array)
    puts canUpdate.inspect
  end
end